golint/golint.go

```diff
@@ -11,15 +11,23 @@ import (
 	"flag"
 	"fmt"
 	"go/build"
-	"io/ioutil"
+	"go/parser"
 	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 
+	"golang.org/x/tools/go/loader"
+
 	"github.com/golang/lint"
 )
 
 var minConfidence = flag.Float64("min_confidence", 0.8, "minimum confidence of a problem to print it")
+var buildTags stringsFlag
+
+func init() {
+	flag.Var(&buildTags, "tags", "a list of build tags to consider satisfied during the build")
```

Comment:
I'm not sure about build tags too. This needs to thought. Ideally we'd lint all possible code, but dragging in the type checker obviously means we need to select between mutually exclusive source files.